tests/load-tests/pkg/journey/handle_collections.go

```diff
@@ -131,6 +134,45 @@ func collectPipelineRunJSONs(f *framework.Framework, dirPath, namespace, applica
 	return nil
 }
 
+func collectApplicationComponentJSONs(f *framework.Framework, dirPath, namespace, application, component string) error {
+	// Only save Application JSON if it has not already been collected (as HandlePerComponentCollection method is called for each component)
+	if _, err := os.Stat(filepath.Join(dirPath, "collected-application-" + application + ".json")); errors.Is(err, os.ErrNotExist) {
```

Comment:
Should this wrapper construct be either dropped or used for component as well?

Also this `"collected-application-" + application + ".json"` is being done two times, so maybe put it into variable?

Reply:
> Should this wrapper construct be either dropped or used for component as well?

I have only done the wrapper logic for the application as the below collect logic (HandlePerComponentCollection) is being called at each component level. So when we create an application with 2 components, then we don't want to capture the same application JSON twice. 

---
> Also this "collected-application-" + application + ".json" is being done two times, so maybe put it into variable?

Yes, will update this change.